code/modules/events/old_vendotron_event/vendor_teleport_event.dm

```diff
@@ -15,44 +15,45 @@
 	teleportVendor()
 
 /datum/event/old_vendotron_teleport/proc/teleportVendor()
-	var/turf/vendT = vendSpawnDecide()
-	fancyEntrance(vendT)
+	var/atom/chosenVend = vendSpawnDecide()
+	fancyEntrance(chosenVend)
 
 /datum/event/old_vendotron_teleport/proc/vendSpawnDecide()
-	var/list/dontSpawnHere = list(
-		/area/derelictparts,
-		/area/solar,
-		/area/assembly,	//Because I don't know what that is
-		/area/shuttle/administration/station,
-		/area/ai_monitored/storage/emergency,
-		/area/arrival,
-		/area/shuttle/escape_pod1,
-		/area/shuttle/escape_pod2,
-		/area/shuttle/escape_pod3,
-		/area/shuttle/escape_pod4,
-		/area/shuttle/escape_pod5,
-		/area/shuttle/prison/,
+	var/list/canReplace = list(
```

Comment:
```suggestion
	var/static/list/canReplace = list(
```
`is_type_in_list` is not supposed to be used with short-lived lists. It's optimized so that future calls of it on the same list will run much faster, at the cost of doing a little more work upfront, which means that using it on a list that's going to be destroyed immediately after performs more work than a for-loop+istype would.

This isn't going to be called a lot, so it doesn't really matter, but making this a static var is free performance, and a good habit.

Reply:
Done.